skiko/src/jvmMain/kotlin/org/jetbrains/skiko/SkiaLayer.jvm.kt

```diff
@@ -95,10 +95,21 @@ actual open class SkiaLayer internal constructor(
     }
 
     override fun removeNotify() {
-        dispose()
+        disposeSkiaResources()
+        isDisposed = true
         super.removeNotify()
     }
 
+    override fun addNotify() {
+        super.addNotify()
+        if (isInited) {
```

Comment:
We have initialization in `init` method, which we happened when window is Displayeable.

I am afraid we can call `addNotify` and `init` at the same time. That can lead to memory leaks and double initialization.

The more right approach will be to remove `if (it.changeFlags and HierarchyEvent.DISPLAYABILITY_CHANGED.toLong() != 0L) {` and call `init` here.

Probably the easiest code will be:
```
override fun addNotify() {
        super.addNotify()
        init()
}


override fun removeNotify() {
        dispose()
        super.addNotify()
}
```

Reply:
No - we shouldn't call `backedLayer.init ()` in case of a "recreation window". All we need to do is recreate the `ContextHandler` and `Redrawer` with the current `renderApi`.

Reply:
> we shouldn't call backedLayer.init () in case of a "recreation window".

Why? Imagine the window is recreated on macOs. In that case `window` property will point to a wrong window.

1. `addNotify` not only called in case of recreation window, but in the usual case.
2. decision of calling or not calling of `backedLayer.init()` shouldn't be based on its implementation.
3. init/dispose symmetry without additional init/dispose will significally simplify the code

Reply:
Check new commit.